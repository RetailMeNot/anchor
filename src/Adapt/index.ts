export { Adapt } from './Adapt.component';
