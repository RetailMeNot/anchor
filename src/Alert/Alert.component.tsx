// REACT
import * as React from 'react';
// VENDOR
import classNames from 'classnames';
import styled, {
    css,
    FlattenInterpolation,
    ThemeProps,
} from '@xstyled/styled-components';
import { th } from '@xstyled/system';
import { transparentize, darken } from 'polished';
// COMPONENTS
import { Close } from '../Icon';
import { Typography } from '../Typography';
import { Button } from '../Button';
import { get } from '../utils/get/get';

export enum AlertTypes {
    success = 'SUCCESS',
    info = 'INFO',
    warning = 'WARNING',
    error = 'ERROR',
}

type AlertType =
    | 'success'
    | 'info'
    | 'warning'
    | 'error'
    | AlertTypes.success
    | AlertTypes.info
    | AlertTypes.warning
    | AlertTypes.error;

interface AlertProps extends React.HTMLAttributes<HTMLDivElement> {
    type: AlertType;
    className?: string;
    message?: string;
    description?: string;
    closable?: boolean;
    icon?: JSX.Element;
    onClose?: (event: Event) => void;
}

const renderAlertTheme = (
    type: AlertType
): FlattenInterpolation<ThemeProps<AlertProps>> => {
    return css`
        ${({ theme }: any) =>
            css({
                backgroundColor: transparentize(
                    0.75,
                    get(theme, `colors.${type.toLowerCase()}`)
                ),
                border: `solid thin ${transparentize(
                    0.4,
                    get(theme, `colors.${type.toLowerCase()}`)
                )}`,
            })};

        .anchor-icon {
            ${({ theme }) =>
                css({
                    color: darken(
                        0.2,
                        get(theme, `colors.${type.toLowerCase()}`)
                    ),
                })}
        }
    `;
};

const StyledAlert = styled('div')<AlertProps>`
    box-sizing: border-box;
    position: relative;
    border-radius: base;
    font-family: ${th('typography.fontFamily')};
    padding: 0.5rem 1.5rem 0.5rem 0.5rem;
    ${({ type }) => renderAlertTheme(type)};

    .anchor-alert-close {
        position: absolute;
        top: 0;
        right: 0;

        &:active,
        &:hover,
        &:focus {
            background-color: transparent;
        }
    }

    h4,
    p {
        line-height: 1.2em;
        margin: 0;
        padding: 0;
    }

    h4 {
        padding-bottom: 0.75rem;
        &:last-child {
            padding-bottom: 0;
        }
    }
`;

const renderMessageAndDescription = ({
    message,
    description,
}: {
    message?: string;
    description?: string;
}) => (
    <>
        {message && <Typography tag="h4">{message}</Typography>}
        {description && (
            <Typography tag="p" color="charcoal" hue="light">
                {description}
            </Typography>
        )}
    </>
);

const StyledIconContainer = styled('div')`
    box-sizing: border-box;
    display: flex;

    .anchor-alert-content {
        flex: 1 1 auto;
    }
    .anchor-icon {
        flex: 0 0 0;
        padding: 0 0.5rem 0 0;
    }
`;

export const Alert = ({
    className,
    children,
    onClose = () => null,
    message,
    description,
    closable,
    icon,
    ...props
}: AlertProps): React.ReactElement<AlertProps> | null => {
    const [isHidden, setIsHidden] = React.useState<boolean>(false);

    return !isHidden ? (
        <StyledAlert
            className={classNames('anchor-alert', className)}
            {...props}
        >
            {closable && (
                <Button
                    className="anchor-alert-close"
                    size="xs"
                    circular={true}
                    variant="minimal"
                    prefix={<Close color="rgba(0, 0, 0, 0.3)" scale="sm" />}
                    onClick={() => setIsHidden(true)}
                />
            )}
            {icon ? (
                <StyledIconContainer>
                    {React.cloneElement(icon)}
                    <div className="anchor-alert-content">
                        {renderMessageAndDescription({ message, description })}
                    </div>
                </StyledIconContainer>
            ) : (
                renderMessageAndDescription({ message, description })
            )}
        </StyledAlert>
    ) : null;
};
