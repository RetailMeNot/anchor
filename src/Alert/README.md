# Alert

The alert component is an inline element to communicate the status of an event. 

## Usage

    import { Alert } from '@retailmenot/anchor';

    const example = () => (
        <Alert type="info" message="This is an alert" />
    );

## Settings/Props

The component can accept the following `props`:

* `type` *('success' | 'info' | warning' | 'error')*: Required. Determines the color treatment of the Alert.
* `className` *(string)*: Sets a top-level class on the Alert component.
* `message` *(string)*: Sets the main message of the Alert.
* `description` *(string)*: Sets the secondary content of the Alert.
* `closable` *(boolean)*: Defaults to **`false`**. Determines whether the Alert component is closable or not.
* `icon` *(JSX.Element)*: An Icon to be rendered within the Alert.
* `onClose` *(() => void)*: An event handler that fires when the Alert is closed.

## Dev Notes

This component doesn't have any exceptional behavior.
