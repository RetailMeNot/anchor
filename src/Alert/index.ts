export { Alert } from './Alert.component';
