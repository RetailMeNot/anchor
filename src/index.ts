export { Avatar } from './Avatar';
export { AutoComplete } from './AutoComplete';
export { Button } from './Button/Button.component';
export { Badge } from './Badge';
export { DropDown } from './DropDown';
export { Container, Flex, Box } from './Grid';
export { Icon } from './Icon';
export { Page, Footer, Header } from './Layout';
export { List, Item as ListItem } from './List';
export { Menu, Item as MenuItem } from './Menu';
export { colors, fonts, sizes, variables, GlobalCSS, NormalizeCSS } from './theme';
