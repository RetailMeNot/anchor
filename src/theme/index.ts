export { default as colors } from './colors.theme';
export { default as fonts } from './fonts.theme';
export { default as sizes } from './sizes.theme';
export { default as variables } from './variables.theme';

export { NormalizeCSS, GlobalCSS } from './GlobalStyles';
