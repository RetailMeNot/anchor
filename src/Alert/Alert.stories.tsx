// REACT
import * as React from 'react';
// STORYBOOK
import { storiesOf } from '@storybook/react';
// VENDOR
import styled, { ThemeProvider } from '@xstyled/styled-components';
import { th } from '@xstyled/system';
import { select, boolean, text } from '@storybook/addon-knobs';
// ANCHOR
import { RootTheme } from '../../src/theme';
import { Alert, AlertTypes } from './Alert.component';
import { InfoOutline } from '../Icon';

const StyledStory = styled('div')`
    padding: 2rem;
    color: text.body;
    font-family: ${th('typography.fontFamily')};
`;

storiesOf('Components/Alert', module)
    .add('Default', () => (
        <ThemeProvider theme={RootTheme}>
            <StyledStory>
                <Alert
                    type={select(
                        'Type',
                        [
                            AlertTypes.warning,
                            AlertTypes.error,
                            AlertTypes.success,
                            AlertTypes.info,
                        ],
                        AlertTypes.warning
                    )}
                    closable={boolean('Closable', true)}
                    message={text('Message', 'This is a message')}
                    description={text('Description', 'This is a description')}
                />
            </StyledStory>
        </ThemeProvider>
    ))
    .add('With Icon', () => (
        <ThemeProvider theme={RootTheme}>
            <StyledStory>
                <Alert
                    type={select(
                        'Type',
                        [
                            AlertTypes.warning,
                            AlertTypes.error,
                            AlertTypes.success,
                            AlertTypes.info,
                        ],
                        AlertTypes.warning
                    )}
                    closable={boolean('Closable', true)}
                    message={text('Message', 'This is a message')}
                    description={text('Description', 'This is a description')}
                    icon={<InfoOutline scale="md" />}
                />
            </StyledStory>
        </ThemeProvider>
    ));
