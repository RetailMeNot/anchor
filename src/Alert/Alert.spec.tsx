// VENDOR
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { ThemeProvider } from '@xstyled/styled-components';
import { shallow, mount } from 'enzyme';
// ANCHOR
import { RootTheme } from '../../src/theme';
import { Alert } from './Alert.component';

describe('Component: Alert', () => {
    it('should be defined.', () => {
        const subject = (
            <ThemeProvider theme={RootTheme}>
                <Alert />
            </ThemeProvider>
        );
        const wrapper = mount(subject);
        const component = shallow(subject);

        expect(subject).toBeDefined();
        expect(wrapper).toBeDefined();
        expect(component).toBeDefined();
        const tree = renderer.create(subject).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
