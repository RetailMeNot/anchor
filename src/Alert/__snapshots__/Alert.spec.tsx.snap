// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Alert should match its snapshot. 1`] = `
Array [
  .c0 {
  box-sizing: border-box;
  position: relative;
  border-radius: 4px;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  padding: 0.5rem 1.5rem 0.5rem 0.5rem;
  background-color: rgba(255,199,97,0.25);
  border: solid thin;
  border-color: rgba(255,199,97,0.6);
}

.c0 .anchor-alert-icon {
  color: #faa100;
}

.c0 .anchor-alert-close {
  position: absolute;
  top: 0;
  right: 0;
}

.c0 .anchor-alert-close:active,
.c0 .anchor-alert-close:hover,
.c0 .anchor-alert-close:focus {
  background-color: transparent;
}

.c0 h4,
.c0 p {
  line-height: 1.2em;
  margin: 0;
  padding: 0;
}

.c0 h4 {
  padding-bottom: 0.75rem;
}

.c0 h4:last-child {
  padding-bottom: 0;
}

.c1 {
  box-sizing: border-box;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  font-size: 16px;
  font-weight: normal;
  line-height: 1.5rem;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: normal;
  font-size: 1rem;
  line-height: 1.5rem;
  margin-bottom: 1rem;
  text-align: inherit;
  text-transform: none;
  color: #595959;
}

.c1 small {
  font-size: 87.5%;
}

<div
    className="c0 anchor-alert"
    title="Test Title"
  >
    <p
      className="c1 anchor-typography"
      color="charcoal"
    >
      Test Description
    </p>
  </div>,
  .c2 {
  display: inline-block;
  height: 1rem;
  width: 1rem;
  line-height: 0;
}

.c0 {
  box-sizing: border-box;
  position: relative;
  border-radius: 4px;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  padding: 0.5rem 1.5rem 0.5rem 0.5rem;
  background-color: rgba(255,199,97,0.25);
  border: solid thin;
  border-color: rgba(255,199,97,0.6);
}

.c0 .anchor-alert-icon {
  color: #faa100;
}

.c0 .anchor-alert-close {
  position: absolute;
  top: 0;
  right: 0;
}

.c0 .anchor-alert-close:active,
.c0 .anchor-alert-close:hover,
.c0 .anchor-alert-close:focus {
  background-color: transparent;
}

.c0 h4,
.c0 p {
  line-height: 1.2em;
  margin: 0;
  padding: 0;
}

.c0 h4 {
  padding-bottom: 0.75rem;
}

.c0 h4:last-child {
  padding-bottom: 0;
}

.c1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 .anchor-alert-content {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.c1 .anchor-icon {
  -webkit-flex: 0 0 0;
  -ms-flex: 0 0 0;
  flex: 0 0 0;
  padding: 0 0.5rem 0 0;
}

.c3 {
  box-sizing: border-box;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  font-size: 16px;
  font-weight: normal;
  line-height: 1.5rem;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: normal;
  font-size: 1rem;
  line-height: 1.5rem;
  margin-bottom: 1rem;
  text-align: inherit;
  text-transform: none;
  color: #595959;
}

.c3 small {
  font-size: 87.5%;
}

<div
    className="c0 anchor-alert"
    title="Test Title"
  >
    <div
      className="c1"
    >
      <span
        className="c2 anchor-icon info anchor-alert-icon"
        scale="md"
      >
        <svg
          height={16}
          viewBox="0 0 16 16"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fillRule="evenodd"
          >
            <circle
              cx="8"
              cy="8"
              fill="currentColor"
              r="8"
            />
            <path
              d="M7.292 7.5h1.416a.5.5 0 0 1 .5.5v3.5a.5.5 0 0 1-.5.5H7.292a.5.5 0 0 1-.5-.5V8a.5.5 0 0 1 .5-.5zM6.75 5.255a1.259 1.259 0 0 1 .371-.883c.115-.113.25-.204.403-.27a1.21 1.21 0 0 1 .962 0c.147.066.278.157.393.27.115.114.205.247.272.4.066.152.099.313.099.483a1.228 1.228 0 0 1-.764 1.15 1.267 1.267 0 0 1-.962 0 1.3 1.3 0 0 1-.403-.261 1.228 1.228 0 0 1-.371-.888z"
              fill="#FFF"
              fillRule="nonzero"
            />
          </g>
        </svg>
      </span>
      <div
        className="anchor-alert-content"
      >
        <p
          className="c3 anchor-typography"
          color="charcoal"
        >
          Test Description
        </p>
      </div>
    </div>
  </div>,
  .c0 {
  box-sizing: border-box;
  position: relative;
  border-radius: 4px;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  padding: 0.5rem 1.5rem 0.5rem 0.5rem;
  background-color: rgba(102,204,255,0.25);
  border: solid thin;
  border-color: rgba(102,204,255,0.6);
}

.c0 .anchor-alert-icon {
  color: #0af;
}

.c0 .anchor-alert-close {
  position: absolute;
  top: 0;
  right: 0;
}

.c0 .anchor-alert-close:active,
.c0 .anchor-alert-close:hover,
.c0 .anchor-alert-close:focus {
  background-color: transparent;
}

.c0 h4,
.c0 p {
  line-height: 1.2em;
  margin: 0;
  padding: 0;
}

.c0 h4 {
  padding-bottom: 0.75rem;
}

.c0 h4:last-child {
  padding-bottom: 0;
}

.c1 {
  box-sizing: border-box;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  font-size: 16px;
  font-weight: normal;
  line-height: 1.5rem;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: normal;
  font-size: 1rem;
  line-height: 1.5rem;
  margin-bottom: 1rem;
  text-align: inherit;
  text-transform: none;
  color: #595959;
}

.c1 small {
  font-size: 87.5%;
}

<div
    className="c0 anchor-alert"
    title="Test Title"
  >
    <p
      className="c1 anchor-typography"
      color="charcoal"
    >
      Test Description
    </p>
  </div>,
  .c2 {
  display: inline-block;
  height: 1rem;
  width: 1rem;
  line-height: 0;
}

.c0 {
  box-sizing: border-box;
  position: relative;
  border-radius: 4px;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  padding: 0.5rem 1.5rem 0.5rem 0.5rem;
  background-color: rgba(102,204,255,0.25);
  border: solid thin;
  border-color: rgba(102,204,255,0.6);
}

.c0 .anchor-alert-icon {
  color: #0af;
}

.c0 .anchor-alert-close {
  position: absolute;
  top: 0;
  right: 0;
}

.c0 .anchor-alert-close:active,
.c0 .anchor-alert-close:hover,
.c0 .anchor-alert-close:focus {
  background-color: transparent;
}

.c0 h4,
.c0 p {
  line-height: 1.2em;
  margin: 0;
  padding: 0;
}

.c0 h4 {
  padding-bottom: 0.75rem;
}

.c0 h4:last-child {
  padding-bottom: 0;
}

.c1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 .anchor-alert-content {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.c1 .anchor-icon {
  -webkit-flex: 0 0 0;
  -ms-flex: 0 0 0;
  flex: 0 0 0;
  padding: 0 0.5rem 0 0;
}

.c3 {
  box-sizing: border-box;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  font-size: 16px;
  font-weight: normal;
  line-height: 1.5rem;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: normal;
  font-size: 1rem;
  line-height: 1.5rem;
  margin-bottom: 1rem;
  text-align: inherit;
  text-transform: none;
  color: #595959;
}

.c3 small {
  font-size: 87.5%;
}

<div
    className="c0 anchor-alert"
    title="Test Title"
  >
    <div
      className="c1"
    >
      <span
        className="c2 anchor-icon info anchor-alert-icon"
        scale="md"
      >
        <svg
          height={16}
          viewBox="0 0 16 16"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fillRule="evenodd"
          >
            <circle
              cx="8"
              cy="8"
              fill="currentColor"
              r="8"
            />
            <path
              d="M7.292 7.5h1.416a.5.5 0 0 1 .5.5v3.5a.5.5 0 0 1-.5.5H7.292a.5.5 0 0 1-.5-.5V8a.5.5 0 0 1 .5-.5zM6.75 5.255a1.259 1.259 0 0 1 .371-.883c.115-.113.25-.204.403-.27a1.21 1.21 0 0 1 .962 0c.147.066.278.157.393.27.115.114.205.247.272.4.066.152.099.313.099.483a1.228 1.228 0 0 1-.764 1.15 1.267 1.267 0 0 1-.962 0 1.3 1.3 0 0 1-.403-.261 1.228 1.228 0 0 1-.371-.888z"
              fill="#FFF"
              fillRule="nonzero"
            />
          </g>
        </svg>
      </span>
      <div
        className="anchor-alert-content"
      >
        <p
          className="c3 anchor-typography"
          color="charcoal"
        >
          Test Description
        </p>
      </div>
    </div>
  </div>,
  .c0 {
  box-sizing: border-box;
  position: relative;
  border-radius: 4px;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  padding: 0.5rem 1.5rem 0.5rem 0.5rem;
  background-color: rgba(20,163,130,0.25);
  border: solid thin;
  border-color: rgba(20,163,130,0.6);
}

.c0 .anchor-alert-icon {
  color: #09483a;
}

.c0 .anchor-alert-close {
  position: absolute;
  top: 0;
  right: 0;
}

.c0 .anchor-alert-close:active,
.c0 .anchor-alert-close:hover,
.c0 .anchor-alert-close:focus {
  background-color: transparent;
}

.c0 h4,
.c0 p {
  line-height: 1.2em;
  margin: 0;
  padding: 0;
}

.c0 h4 {
  padding-bottom: 0.75rem;
}

.c0 h4:last-child {
  padding-bottom: 0;
}

.c1 {
  box-sizing: border-box;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  font-size: 16px;
  font-weight: normal;
  line-height: 1.5rem;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: normal;
  font-size: 1rem;
  line-height: 1.5rem;
  margin-bottom: 1rem;
  text-align: inherit;
  text-transform: none;
  color: #595959;
}

.c1 small {
  font-size: 87.5%;
}

<div
    className="c0 anchor-alert"
    title="Test Title"
  >
    <p
      className="c1 anchor-typography"
      color="charcoal"
    >
      Test Description
    </p>
  </div>,
  .c2 {
  display: inline-block;
  height: 1rem;
  width: 1rem;
  line-height: 0;
}

.c0 {
  box-sizing: border-box;
  position: relative;
  border-radius: 4px;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  padding: 0.5rem 1.5rem 0.5rem 0.5rem;
  background-color: rgba(20,163,130,0.25);
  border: solid thin;
  border-color: rgba(20,163,130,0.6);
}

.c0 .anchor-alert-icon {
  color: #09483a;
}

.c0 .anchor-alert-close {
  position: absolute;
  top: 0;
  right: 0;
}

.c0 .anchor-alert-close:active,
.c0 .anchor-alert-close:hover,
.c0 .anchor-alert-close:focus {
  background-color: transparent;
}

.c0 h4,
.c0 p {
  line-height: 1.2em;
  margin: 0;
  padding: 0;
}

.c0 h4 {
  padding-bottom: 0.75rem;
}

.c0 h4:last-child {
  padding-bottom: 0;
}

.c1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 .anchor-alert-content {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.c1 .anchor-icon {
  -webkit-flex: 0 0 0;
  -ms-flex: 0 0 0;
  flex: 0 0 0;
  padding: 0 0.5rem 0 0;
}

.c3 {
  box-sizing: border-box;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  font-size: 16px;
  font-weight: normal;
  line-height: 1.5rem;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: normal;
  font-size: 1rem;
  line-height: 1.5rem;
  margin-bottom: 1rem;
  text-align: inherit;
  text-transform: none;
  color: #595959;
}

.c3 small {
  font-size: 87.5%;
}

<div
    className="c0 anchor-alert"
    title="Test Title"
  >
    <div
      className="c1"
    >
      <span
        className="c2 anchor-icon check anchor-alert-icon"
        scale="md"
      >
        <svg
          height={16}
          viewBox="0 0 16 16"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fillRule="evenodd"
          >
            <path
              d="M0 0h16v16H0z"
            />
            <path
              d="M14.627 5.133l-7.637 7.5a1.282 1.282 0 0 1-1.8 0l-3.817-3.75a1.235 1.235 0 0 1 0-1.768 1.288 1.288 0 0 1 1.799 0L6.09 9.982l6.737-6.617a1.291 1.291 0 0 1 1.8 0c.496.488.496 1.28 0 1.768"
              fill="currentColor"
            />
          </g>
        </svg>
      </span>
      <div
        className="anchor-alert-content"
      >
        <p
          className="c3 anchor-typography"
          color="charcoal"
        >
          Test Description
        </p>
      </div>
    </div>
  </div>,
  .c0 {
  box-sizing: border-box;
  position: relative;
  border-radius: 4px;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  padding: 0.5rem 1.5rem 0.5rem 0.5rem;
  background-color: rgba(204,59,59,0.25);
  border: solid thin;
  border-color: rgba(204,59,59,0.6);
}

.c0 .anchor-alert-icon {
  color: #802121;
}

.c0 .anchor-alert-close {
  position: absolute;
  top: 0;
  right: 0;
}

.c0 .anchor-alert-close:active,
.c0 .anchor-alert-close:hover,
.c0 .anchor-alert-close:focus {
  background-color: transparent;
}

.c0 h4,
.c0 p {
  line-height: 1.2em;
  margin: 0;
  padding: 0;
}

.c0 h4 {
  padding-bottom: 0.75rem;
}

.c0 h4:last-child {
  padding-bottom: 0;
}

.c1 {
  box-sizing: border-box;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  font-size: 16px;
  font-weight: normal;
  line-height: 1.5rem;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: normal;
  font-size: 1rem;
  line-height: 1.5rem;
  margin-bottom: 1rem;
  text-align: inherit;
  text-transform: none;
  color: #595959;
}

.c1 small {
  font-size: 87.5%;
}

<div
    className="c0 anchor-alert"
    title="Test Title"
  >
    <p
      className="c1 anchor-typography"
      color="charcoal"
    >
      Test Description
    </p>
  </div>,
  .c2 {
  display: inline-block;
  height: 1rem;
  width: 1rem;
  line-height: 0;
}

.c0 {
  box-sizing: border-box;
  position: relative;
  border-radius: 4px;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  padding: 0.5rem 1.5rem 0.5rem 0.5rem;
  background-color: rgba(204,59,59,0.25);
  border: solid thin;
  border-color: rgba(204,59,59,0.6);
}

.c0 .anchor-alert-icon {
  color: #802121;
}

.c0 .anchor-alert-close {
  position: absolute;
  top: 0;
  right: 0;
}

.c0 .anchor-alert-close:active,
.c0 .anchor-alert-close:hover,
.c0 .anchor-alert-close:focus {
  background-color: transparent;
}

.c0 h4,
.c0 p {
  line-height: 1.2em;
  margin: 0;
  padding: 0;
}

.c0 h4 {
  padding-bottom: 0.75rem;
}

.c0 h4:last-child {
  padding-bottom: 0;
}

.c1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 .anchor-alert-content {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.c1 .anchor-icon {
  -webkit-flex: 0 0 0;
  -ms-flex: 0 0 0;
  flex: 0 0 0;
  padding: 0 0.5rem 0 0;
}

.c3 {
  box-sizing: border-box;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  font-size: 16px;
  font-weight: normal;
  line-height: 1.5rem;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: normal;
  font-size: 1rem;
  line-height: 1.5rem;
  margin-bottom: 1rem;
  text-align: inherit;
  text-transform: none;
  color: #595959;
}

.c3 small {
  font-size: 87.5%;
}

<div
    className="c0 anchor-alert"
    title="Test Title"
  >
    <div
      className="c1"
    >
      <span
        className="c2 anchor-icon error anchor-alert-icon"
        scale="md"
      >
        <svg
          height={16}
          viewBox="0 0 16 16"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fillRule="evenodd"
          >
            <path
              d="M0 0h16v16H0z"
            />
            <path
              d="M15.797 12.757c.133.23.203.49.203.756 0 .821-.652 1.487-1.457 1.487H1.457c-.271 0-.537-.077-.768-.223a1.504 1.504 0 0 1-.47-2.047L6.921 1.703a1.44 1.44 0 0 1 2.493.027l6.383 11.027z"
              fill="currentColor"
            />
            <path
              d="M7.133 9.377V5.5a.5.5 0 0 1 .5-.5h.723a.5.5 0 0 1 .5.5v3.877a.5.5 0 0 1-.5.5h-.723a.5.5 0 0 1-.5-.5zM9 11.451a.983.983 0 0 1-.077.388.902.902 0 0 1-.215.311c-.093.086-.2.155-.323.207a.995.995 0 0 1-.39.078c-.137 0-.265-.024-.385-.073a.92.92 0 0 1-.313-.207 1.095 1.095 0 0 1-.215-.31.958.958 0 0 1-.005-.761.973.973 0 0 1 .528-.528.995.995 0 0 1 .39-.078c.137 0 .267.026.39.077.123.052.23.121.323.208a.958.958 0 0 1 .292.689z"
              fill="#FFF"
              fillRule="nonzero"
            />
          </g>
        </svg>
      </span>
      <div
        className="anchor-alert-content"
      >
        <p
          className="c3 anchor-typography"
          color="charcoal"
        >
          Test Description
        </p>
      </div>
    </div>
  </div>,
]
`;
